test/cli/test_new.rb

```diff
@@ -0,0 +1,257 @@
+      when "unstructured"
+        output =  "text:#{text}\n"
+      else
+        # should have already bailed in the previous case statement...
+        p "wtf, #{type}"
+        # can't test an unknown or custom type
+        next
+      end
+
+      # check record:
+      if @@debugging
+        p "checking #{type}"
+        p "expect: #{output}"
+        p "actual: " + last_command_stopped.output
+      end
+      assert_equal "#{output}", last_command_stopped.output
+
+      # remove record:
+      run_command_and_stop "kbsecret rm #{label}"
+    end
+  end
+
+  def test_login
+    label = "test-login"
+    username = "user"
+    password = "pass"
+
+    # this would be better, if only it worked:
+    # run_command "kbsecret new login -x #{label}"
+    # type "#{username}"
+    # type "#{password}"
+    # close_input
+    # stop_all_commands
+
+    # NOTE: bash -c workaround
+    # create login:
+    run_command_and_stop "bash -c 'kbsecret new login -x #{label} <<< #{username}:#{password}'"
+
+    # retrieve login:
+    run_command_and_stop "kbsecret login -x #{label}"
+    assert_equal "#{label}:#{username}:#{password}", last_command_stopped.output.chomp
+
+    # retrieve pass:
+    run_command_and_stop "kbsecret pass #{label}"
+    assert_equal "#{password}", last_command_started.output.chomp
+
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+
+    # fail to retreive removed login:
+    run_command "kbsecret pass #{label}"
+    stop_all_commands
+    assert_match(/No such record/, last_command_started.stderr.chomp)
+  end
+
+  def test_force
+    label = "test-login-force"
+    username = "user"
+    password = "pass"
+    username2 = "user2"
+    password2 = "pass2"
+
+    # NOTE: bash -c workaround
+    # create login:
+    run_command_and_stop "bash -c 'kbsecret new login -x #{label} <<< #{username}:#{password}'"
+
+    # retrieve login:
+    run_command_and_stop "kbsecret login -x #{label}"
+    assert_equal "#{label}:#{username}:#{password}", last_command_stopped.output.chomp
+
+    # NOTE: bash -c workaround
+    # force overwrite of login:
+    run_command_and_stop "bash -c 'kbsecret new login --force -x #{label} <<< #{username2}:#{password2}'"
+
+    # retrieve overwritten login:
+    run_command_and_stop "kbsecret login -x #{label}"
+    assert_equal "#{label}:#{username2}:#{password2}", last_command_stopped.output.chomp
+
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
+  end
+
+  def test_separator
+    label = "test-separator"
+    username = "user"
+    password = "pass"
+
+    # NOTE: bash -c workaround
+    # create login:
+    run_command_and_stop "bash -c 'kbsecret new login -i ^ -x #{label} <<< #{username}^#{password}'"
+
+    # retrieve login:
+    run_command_and_stop "kbsecret login -x #{label}"
+    assert_equal "#{label}:#{username}:#{password}", last_command_stopped.output.chomp
+
+    # remove login:
+    run_command_and_stop "kbsecret rm #{label}"
```

Comment:
Ruby methods can have an `ensure` block at the end, and this would probably be a good use of one:

```ruby
ensure
  run_command_and_stop "kbsecret rm #{label}"
end # end of method is the same as end of ensure
```